tests/everflow/everflow_test_utilities.py

```diff
@@ -47,111 +48,148 @@
+
+    for ns, neigh_set in upstream_neigh_namespace_map.items():
+        if len(neigh_set) < 2:
+            upstream_ports_namespace.remove(ns)
+    
+    for ns, neigh_set in downstream_neigh_namespace_map.items():
+        if len(neigh_set) < 2:
+            downstream_ports_namespace.remove(ns)
+
+    if not upstream_ports_namespace or not downstream_ports_namespace:
+        pytest.skip("Not enough ports for upstream or downstream neighbors to run this test")
+   
     if 't1' in topo:
         # Set of downstream ports only Namespace
         downstream_only_namespace = downstream_ports_namespace.difference(upstream_ports_namespace)
         # Set of upstream ports only Namespace
         upstream_only_namespace = upstream_ports_namespace.difference(downstream_ports_namespace)
         # Randomly choose from downstream_only Namespace if present else just use first one
         downstream_namespace = random.choice(tuple(downstream_only_namespace)) \
-            if downstream_only_namespace else tuple(downstream_ports_namespace)[0]
+            if downstream_only_namespace else random.choice(tuple(downstream_ports_namespace))
         # Randomly choose from upstream_only Namespace if present else just use first one
         upstream_namespace = random.choice(tuple(upstream_only_namespace)) \
-            if upstream_only_namespace else tuple(upstream_ports_namespace)[0]
+            if upstream_only_namespace else random.choice(tuple(upstream_ports_namespace))
     else:
         # Use the default namespace
-        downstream_namespace = tuple(downstream_ports_namespace)[0]
-        upstream_namespace = tuple(upstream_ports_namespace)[0]
+        downstream_namespace = random.choice(tuple(downstream_ports_namespace))
+        upstream_namespace = random.choice(tuple(upstream_ports_namespace))
 
     downstream_ports = downstream_ports_namespace_map[downstream_namespace]
     upstream_ports = upstream_ports_namespace_map[upstream_namespace]
 
-    switch_capabilities = switch_capability_facts["switch_capabilities"]["switch"]
-    acl_capabilities = acl_capability_facts["acl_capabilities"]
+    random.shuffle(downstream_ports)
+    random.shuffle(upstream_ports)
+
+
+    upstream_switch_capabilities = upstream_switch_capability_facts["switch_capabilities"]["switch"]
+    upstream_acl_capabilities = upstream_acl_capability_facts["acl_capabilities"]
+
+    downstream_switch_capabilities = downstream_switch_capability_facts["switch_capabilities"]["switch"]
+    downstream_acl_capabilities = downstream_acl_capability_facts["acl_capabilities"]
+
 
-    test_mirror_v4 = switch_capabilities["MIRROR"] == "true"
-    test_mirror_v6 = switch_capabilities["MIRRORV6"] == "true"
+    test_mirror_v4 = upstream_switch_capabilities["MIRROR"] == "true" and downstream_switch_capabilities["MIRROR"] == "true"
+    test_mirror_v6 = upstream_switch_capabilities["MIRRORV6"] == "true" and  downstream_switch_capabilities["MIRRORV6"] == "true"
```

Comment:
For T0 device, what's the `downstream_switch_capabilities["MIRROR"]` ?

Reply:
for T0 upStream and downStreamDut are same so both point to same capabilities. This is just to keep code geneic and not have `if..else`